File: Java/CesiumLanguageWriter/translatedSrc/cesiumlanguagewriter/advanced/ICesiumUrlResolver.java
package cesiumlanguagewriter.advanced;


import agi.foundation.compatibility.*;
import cesiumlanguagewriter.*;

/**
 *  
 A URL resolver transforms URLs into another form for inclusion in a CZML document.
 For example,  {@link CachingCesiumUrlResolver} downloads files and converts them to data URIs,
 while  {@link PassThroughCesiumUrlResolver} leaves URLs unchanged.
 

 */
public interface ICesiumUrlResolver {
	/**
	 *  
	Resolves a URL, producing a new URL for inclusion in a CZML document.
	
	
	

	 * @param url The source URL.
	 * @return A URL suitable for CZML.
	 */
	String resolveUrl(String url);
}

File: DotNet/CesiumLanguageWriter/Advanced/ICesiumUrlResolver.cs
namespace CesiumLanguageWriter.Advanced
{
    /// <summary>
    /// A URL resolver transforms URLs into another form for inclusion in a CZML document.
    /// For example, <see cref="CachingCesiumUrlResolver"/> downloads files and converts them to data URIs,
    /// while <see cref="PassThroughCesiumUrlResolver"/> leaves URLs unchanged.
    /// </summary>
    public interface ICesiumUrlResolver
    {
        /// <summary>
        /// Resolves a URL, producing a new URL for inclusion in a CZML document.
        /// </summary>
        /// <param name="url">The source URL.</param>
        /// <returns>A URL suitable for CZML.</returns>
        string ResolveUrl(string url);
    }
}